src/components/common/WalletWarning.tsx

```diff
@@ -1,23 +1,60 @@
 import ArrowForwardIcon from '@mui/icons-material/ArrowForward';
 import InfoOutlinedIcon from '@mui/icons-material/InfoOutlined';
 import { Alert, Box, Snackbar, Typography, useTheme } from '@mui/material';
-import React from 'react';
+import React, { useEffect } from 'react';
 import { useWallet } from '../../contexts/wallet';
+import { useStore } from '../../store/store';
+import { toCompactAddress } from '../../utils/formatter';
 import { OpaqueButton } from './OpaqueButton';
+import { Row } from './Row';
 
 export const WalletWarning = () => {
   const theme = useTheme();
-  const { connect, connected } = useWallet();
+  const { connect, connected, walletAddress } = useWallet();
+  const isFunded = useStore((state) => state.isFunded);
+  const loadUserData = useStore((state) => state.loadUserData);
+
   const [openCon, setOpenCon] = React.useState(false);
 
   const handleSnackClose = () => {
     setOpenCon(false);
   };
 
+  useEffect(() => {
+    if (connected && !isFunded) {
+      loadUserData(walletAddress);
+      const refreshInterval = setInterval(async () => {
+        await loadUserData(walletAddress);
+      }, 3 * 1000);
+      return () => clearInterval(refreshInterval);
+    }
+  }, [loadUserData, connected, isFunded, walletAddress]);
```

Comment:
using a separate loading function as this avoids hammering the RPC for blend data.

It will fail immediately if we are unable to fetch the horizon account on the first API call made (which will fail until the account is funded)